test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                                    tokenLimitPerAddress: value.mul(10),
+                                    tokenLimit: BigNumber.from(0),
+                                });
+                        });
+
+                        await this.subject
+                            .connect(this.admin)
+                            .addDepositorsToAllowlist([this.deployer.address]);
+                        await this.weth
+                            .connect(this.deployer)
+                            .approve(this.subject.address, value);
+                        await this.usdc
+                            .connect(this.deployer)
+                            .approve(this.subject.address, value);
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit([value, value], BigNumber.from(1), [])
+                        ).to.be.revertedWith(Exceptions.LIMIT_OVERFLOW);
+                    });
+                });
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await this.subject
+                        .connect(this.admin)
+                        .addDepositorsToAllowlist([this.deployer.address]);
+                    await withSigner(this.deployer.address, async (signer) => {
+                        await this.weth
+                            .connect(signer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await this.usdc
+                            .connect(signer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await expect(
+                            this.subject
+                                .connect(signer)
+                                .deposit(
+                                    [
+                                        BigNumber.from(10001),
+                                        BigNumber.from(10001),
+                                    ],
+                                    BigNumber.from(1),
+                                    []
+                                )
+                        ).to.not.be.reverted;
+                    });
+                });
+            });
+        });
+
+        describe("#withdraw", () => {
+            it("emits Withdraw event", async () => {
+                await this.subject
+                    .connect(this.admin)
+                    .addDepositorsToAllowlist([this.deployer.address]);
+                await this.weth
+                    .connect(this.deployer)
+                    .approve(this.subject.address, BigNumber.from(10001));
```

Comment:
You could also move approve into beforeEach section 
Just to make sure, that every test will pass, give an approve by Uint256Max